lib/cadet/course/course.ex

```diff
@@ -57,6 +56,24 @@ defmodule Cadet.Course do
     end
   end
 
+  @doc """
+  Create group entity with given name
+  """
+  def create_group(name, leader, mentor) do
+    %Group{}
+    |> Group.changeset(Map.get(leader, :id), Map.get(mentor, :id), %{name: name})
+    |> Repo.insert()
+  end
+
+  @doc """
+  Adds given student to given group
+  """
+  def add_student_to_group(group = %Group{}, student = %User{role: :student}) do
+    student
+    |> User.student_changeset(Map.get(group, :id))
+    |> Repo.update!()
```

Comment:
Try not to use the bang version of functions unless you can make sure they won't crash.

Also for consistency since in `create_group/3` you're using the non-bang version